src/main/java/com/aws/greengrass/util/orchestration/SystemServiceUtilsFactory.java

```diff
@@ -41,8 +43,41 @@ public synchronized SystemServiceUtils getInstance() {
                 return context.get(SystemdUtils.class);
             }
         } catch (IOException e) {
-            logger.atError().log("Unable to determine init process type");
+            logger.atDebug().log("No systemd is detected");
         }
+
+        try {
+            String systemServiceProcessName = getProcessName(1);
```

Comment:
can you detect this another way, similar to what we do with systemd can we infer that procd is in use without actually trying to read pid 1?

Reply:
The `procd` is a script instead of an executable file located in the file system. So it can't be detected in the same way as systemd. Besides, the solution has a bug. How can we tell which system management is used if there is a `systemd` in `/sbin/init` in some specific OpenWRT port? From my point of view, we should read pid 1 in a Linux-like platform to detect which system management is because that process is actually used.

Reply:
You can check that it exists or not, right? Not that it is executable

Reply:
Consider this situation. If an OpenWRT port has a `systemd` in `/sbin/init`. We can't say it uses `systemd` to manage its system services. On the other way, if there is a `procd` script in some place of init process, we can't say it uses `procd` to manage system services because it might actually use `systemd`. The safest way is to check pid 1 to know which one is actually being used.

Reply:
I don't think that's a real concern unless you can point to an example of that in reality. Why would a system (and especially openwrt) have both systemd and procd?

Reply:
Indeed, it totally makes sense to have only one system service management. I'll check how to detect procd without using read pid 1 status.